client/rpc/client.go

```diff
@@ -50,6 +51,58 @@ func NewRpcClient(rpcServerUrl string, myAddress types.Address, chainId *big.Int
 	return c
 }
 
+// CreateLedger creates a new ledger channel
+func (rc *RpcClient) CreateVirtual(intermediaries []types.Address, counterparty types.Address, ChallengeDuration uint32, outcome outcome.Exit) virtualfund.ObjectiveResponse {
+
+	objReq := virtualfund.NewObjectiveRequest(
+		intermediaries,
+		counterparty,
+		100,
+		outcome,
+		uint64(rand.Float64()), // TODO: Since numeric fields get converted to a float64 in transit we need to prevent overflow
+		common.Address{})
+
+	// Create a channel and store it in the responses map
+	// We will use this channel to wait for the response
+	resRec := make(chan interface{})
+	rc.responses.Store(string(objReq.Id(rc.myAddress, rc.chainId)), resRec)
+
+	requestId := rand.Uint64()
+	rc.idsToMethods.Store(string(fmt.Sprintf("%d", requestId)), serde.VirtualFundRequestMethod)
+
+	message := serde.NewJsonRpcRequest(requestId, serde.VirtualFundRequestMethod, objReq)
+	data, err := json.Marshal(message)
+	if err != nil {
+		panic("Could not marshal direct fund request")
+	}
+	rc.nts.SendMessage(serde.VirtualFundRequestMethod, data)
+
+	objRes := <-resRec
+	return objRes.(virtualfund.ObjectiveResponse)
```

Comment:
Yeah I started investigating a generic function to handle this but didn't get something working. I'll take a second look at this.

Reply:
I'm struggling to implement this nicely with generics. Since I think we're looking at simplifying the request handlers in #1091 we may want to try refactoring this code then.